File: src/main/java/at/hrechny/predictionsbot/database/converter/ZoneIdConverter.java
package at.hrechny.predictionsbot.database.converter;

import java.time.ZoneId;
import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
public class ZoneIdConverter implements AttributeConverter<ZoneId, String> {

  @Override
  public String convertToDatabaseColumn(ZoneId attribute) {
    return attribute.getId();
  }

  @Override
  public ZoneId convertToEntityAttribute(String dbData) {
    return ZoneId.of( dbData );
  }

}

File: src/main/kotlin/at/hrechny/predictionsbot/database/converter/ZoneIdConverter.kt
package at.hrechny.predictionsbot.database.converter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.time.ZoneId

@Converter
class ZoneIdConverter : AttributeConverter<ZoneId, String> {
    override fun convertToDatabaseColumn(attribute: ZoneId?): String? = attribute?.id

    override fun convertToEntityAttribute(dbData: String?): ZoneId? = dbData?.let(ZoneId::of)
}
